### ISSUE TITLE
Raise a warning when a docstring is inherited but the type was changed in the child class

### ISSUE THREAD

**Issue by vepain** [PyCQA/pylint#4916]
### Current problem

For tools like doc builder `Sphinx` and editors' docstring parser, docstring of method inherited from parent class method (abstract or concrete) is not required. In fact, the tools use the docstring of the parent's method.

At the opposite of static linter (*e.g.* `Pydocstyle`), `Pylint` does not raise a warning of missing docstring if the parent's method has one, and you don't need to add some external package (and configs) to simulate the same for static linters (see [Pydocstyle issue 309](https://github.com/PyCQA/pydocstyle/issues/309)).

I think it is a great idea to keep this non-warning feature. In fact:
- docs builder Sphinx and editors' docstring parser works fine without the repeated docstring in the child method
- it respects the **minimal changes principle** in case of docstring modifications (you have to just change the parent's method docstring)

But I think `Pylint` is **not enough specific**: the need of child's docstring modification is sometimes necessary, and `Pylint` will not raise a warning.

As an example, let suppose docstring covention is Numpy:
```python
from abc import abstractmethod


class ParentA:
    """Parent class A."""


class ChildA(ParentA):
    """Child class A."""


class ParentB:
    """Parent class B: can be concrete or abstract class."""

    def __init__(self):
        self._attr: ParentA = ParentA()

    @abstractmethod
    def attr(self) -> ParentA:
        """Return the attribute.

        Returns
        -------
        ParentA
            The attribute attr
        """
        raise NotImplementedError


class ChildB(ParentB):
    """Concrete class ChildB."""

    def __init__(self):
        self._attr: ChildA = ChildA()

    def attr(self) -> ChildA:
        # Pylint will not raise any warning here because docstring of ParentB.attr exists
        return self._attr
```

So the docstring in Sphinx of `ChildB.attr` method will be the one of `ParentB.attr`
But the two methods don't return the same object type: the first `ChildA` type, the second `ParentA` type.

### Desired solution

I suggest `Pylint` raises a warning in that case like:
> "Return type in child method docstring is not enough specific"

So that it will force the user to re-write docstring of the child method like this:

```python
def attr(self) -> ChildA:
  """Return the attribute.

  Returns
  -----------
  ChildA
      The attribute attr
  """
  return self._attr
```

**But** `Pylint` should make sure that *the docstring elements of the parent and child methods that should not change remain the same*.

To conclude, this is what I suggest (with python color, just because it is better than text):
```python
if 'no docstring':
  if 'method is not overriding any method':
    # Warning: missing doc
  else:
    if 'type signatures of parameters or returns have changed':
      # Warning: you have to update overriding method's docstring
else:
  if 'type signatures of parameters or returns have not changed in overriding method':
    # Warning: unnecessary repeated docstring
  else:
    if 'dosctring of overriding and overridden methods are the same':
      # Warning: signatures not enough specific
```

### Additional context

- Some changes may modify the similar warning and errors table between linters in [Pylint issue 3512](https://github.com/PyCQA/pylint/issues/3512)

Versions
----------
- Python 3.9
- Pylint 3.0.0
- Pydocstyle 6.1.1
- Sphinx-build 4.1.2

### Pylint extension

```ini
load-plugins =
    pylint.extensions.emptystring,
    pylint.extensions.comparetozero,
    pylint.extensions.bad_builtin,
    pylint.extensions.docstyle,
    pylint.extensions.redefined_variable_type,
    pylint.extensions.overlapping_exceptions,
    pylint.extensions.docparams
```

Thank you for your amazing works !
